grails-app/utils/org/pih/warehouse/LocalizationUtil.groovy

```diff
@@ -26,9 +26,41 @@ class LocalizationUtil {
         return localizationService.currentLocale
     }
 
+    /**
+     * Convenience util to handle country specific language (for example Mexican Spanish). As we
+     * used to provide supported locales in config as a language code in ISO 639 alpha-2 or
+     * ISO 639 alpha-3 format, to support country specific languages we have to add a country
+     * code in ISO 3166 alpha-2 or UN M.49 numeric-3 format to it.
+     * As there is no Locale constructor that handles `<languageCode>_<countryCode>` for now we
+     * have to handle it here. If it is a country specific language, then the <languageCode>
+     * will be used to determine the fallback (default) locale for missing translations.
+     * TODO: Improve the supportedLocales config structure to fully support all locale options.
+     *       (see context here: https://pihemr.atlassian.net/browse/OBPIH-6410?focusedCommentId=153567)
+     * @param localeCode String with locale code represented as single language code (ex.: "en", "es"),
+     *                   or combined with country code (ex.: "en_GB", "es_MX").
+     * @return Locale
+     * @exception NullPointerException (Locale throws NPE when all params passed to constructor are null)
+     * */
+    static Locale getLocale(String localeCode) {
+        Locale locale
+
+        if (localeCode?.contains("_")) {
+            String[] localeCodeParts = localeCode.split("_")
+            String language = localeCodeParts[0]
+            String country = localeCodeParts[1]
+            locale = new Locale(language, country)
+            locale.setDefault(new Locale(language))
+            return locale
+        }
+
+        locale = new Locale(localeCode)
+        locale.setDefault(new Locale(Holders.config.openboxes.locale.defaultLocale ?: "en"))
+        return locale
+    }
+
     static List<Locale> getSupportedLocales() {
```

Comment:
Since locales are static, we could consider caching them in a map (maybe in the LocalizationService) so that we don't need to reconstruct them every time we do a lookup. Currently, every time we call this method we're re-initializing all of the Locale objects. That's not a big deal since it's not really a lot of computation work, but still a good practice IMO.

If we wanted to get really fancy, we could even build that map on app startup based on the locales we have defined in our properties.